language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeAnnotationDeclConstantCodeAction.java

```diff
@@ -0,0 +1,110 @@
+import io.ballerina.compiler.syntax.tree.AnnotationDeclarationNode;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.codeaction.CodeActionNodeValidator;
+import org.ballerinalang.langserver.codeaction.CodeActionUtil;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action to make annotation declaration constant.
+ *
+ * @since 2201.1.1
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class MakeAnnotationDeclConstantCodeAction extends AbstractCodeActionProvider {
+    public static final String NAME = "Make Annotation Declaration Constant";
+    public static final String DIAGNOSTIC_CODE = "BCE2638";
+
+    @Override
+    public boolean validate(Diagnostic diagnostic,
+                            DiagBasedPositionDetails positionDetails,
+                            CodeActionContext context) {
+        if (!DIAGNOSTIC_CODE.equals(diagnostic.diagnosticInfo().code()) || context.currentSyntaxTree().isEmpty() ||
+                context.currentSemanticModel().isEmpty()) {
+            return false;
+        }
+
+        SyntaxTree syntaxTree = context.currentSyntaxTree().orElseThrow();
+        NonTerminalNode matchedNode = CommonUtil.findNode(new Range(context.cursorPosition(),
+                context.cursorPosition()), syntaxTree);
+        return CodeActionNodeValidator.validate(matchedNode);
+    }
+    
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    DiagBasedPositionDetails positionDetails,
+                                                    CodeActionContext context) {
+        Optional<Pair<Symbol, Path>> symbolAndPath = CodeActionUtil.findSymbolAndPathForDiagnosticRange(diagnostic, 
+                context);
+        if (symbolAndPath.isEmpty()) {
+           return Collections.emptyList();
+        } 
+        Symbol symbol = symbolAndPath.get().getLeft();
+        Path filePath = symbolAndPath.get().getRight();
+        
+        URI uri = filePath.toUri();
+        Optional<NonTerminalNode> node = CommonUtil.findNode(symbol,
```

Comment:
Instead, you can use the `findNode()` method with the `diagnostic` location to get the node. No need to get the symbol and find node from that. The diagnostic's location is the line range of the annotation decl node